src/libraries/System.Private.CoreLib/src/System/IO/StreamReader.cs

```diff
@@ -865,65 +865,73 @@ private int ReadBuffer(Span<char> userBuffer, out bool readToUserBuffer)
 
-                        if (sb != null)
-                        {
-                            sb.Append(tmpCharBuffer, tmpCharPos, i - tmpCharPos);
-                            s = sb.ToString();
-                        }
-                        else
+                        // Note the following common line feed chars:
+                        // \n - UNIX   \r\n - DOS   \r - Mac
+                        if (ch is '\r' or '\n')
                         {
-                            s = new string(tmpCharBuffer, tmpCharPos, i - tmpCharPos);
-                        }
+                            string s;
 
-                        _charPos = tmpCharPos = i + 1;
+                            if (sb != null)
+                            {
+                                sb.Append(tmpCharBuffer, tmpCharPos, i - tmpCharPos);
+                                s = sb.ToString();
+                            }
+                            else
+                            {
+                                s = new string(tmpCharBuffer, tmpCharPos, i - tmpCharPos);
+                            }
 
-                        if (ch == '\r' && (tmpCharPos < tmpCharLen || (await ReadBufferAsync(CancellationToken.None).ConfigureAwait(false)) > 0))
-                        {
-                            tmpCharPos = _charPos;
-                            if (_charBuffer[tmpCharPos] == '\n')
+                            _charPos = tmpCharPos = i + 1;
+
+                            if (ch == '\r' && (tmpCharPos < tmpCharLen || (await ReadBufferAsync(CancellationToken.None).ConfigureAwait(false)) > 0))
                             {
-                                _charPos = ++tmpCharPos;
+                                tmpCharPos = _charPos;
+                                if (_charBuffer[tmpCharPos] == '\n')
+                                {
+                                    _charPos = ++tmpCharPos;
+                                }
                             }
+
+                            return s;
                         }
 
-                        return s;
-                    }
+                        i++;
+                    } while (i < tmpCharLen);
 
-                    i++;
-                } while (i < tmpCharLen);
+                    i = tmpCharLen - tmpCharPos;
+                    sb ??= new StringBuilder(i + 80);
+                    sb.Append(tmpCharBuffer, tmpCharPos, i);
+                } while (await ReadBufferAsync(CancellationToken.None).ConfigureAwait(false) > 0);
 
-                i = tmpCharLen - tmpCharPos;
-                sb ??= new StringBuilder(i + 80);
-                sb.Append(tmpCharBuffer, tmpCharPos, i);
-            } while (await ReadBufferAsync(CancellationToken.None).ConfigureAwait(false) > 0);
+                return sb.ToString();
+            }
 
-            return sb.ToString();
+            return _charPos == _charLen ? ReadInternalFast() : ReadInternalLoop();
```

Comment:
nit: in this repo we usually put local methods at the bottom and just call them from the top

Reply:
Done